src/components/InventoryGroups/Modals/RemoveHostsFromGroupModal.js

```diff
@@ -7,62 +7,78 @@ import { removeHostsFromGroup } from '../utils/api';
 import componentTypes from '@data-driven-forms/react-form-renderer/component-types';
 import { Text } from '@patternfly/react-core';
 
-const schema = (groupName, hosts) => ({
-  fields: [
-    {
-      component: componentTypes.PLAIN_TEXT,
-      name: 'warning-message',
-      label:
-        hosts.length === 1 ? (
-          <Text>
-            <strong>{hosts[0].display_name}</strong> will no longer be part of{' '}
-            <strong>{groupName}</strong> and its configuration will be impacted.
-          </Text>
-        ) : (
-          <Text>
-            <strong>{hosts.length}</strong> systems will no longer be part of{' '}
-            <strong>{groupName}</strong> and their configuration will be
-            impacted.
-          </Text>
-        ),
-    },
-  ],
-});
+const schema = (hosts) => {
+  const hostsInGroup = hosts.filter(({ groups }) => groups.length > 0); // selection can contain ungroupped hosts
+  const groupName = hostsInGroup[0].groups[0].name;
+
+  return {
+    fields: [
+      {
+        component: componentTypes.PLAIN_TEXT,
+        name: 'warning-message',
+        label:
+          hostsInGroup.length === 1 ? (
```

Comment:
These lines can also be shortened using the template literal with the ternary operator 👍🏼 
Something like this:
```
const numOfSystems = hostsInGroup.length;
const systemName = hostsInGroup[0].display_name;
label:
  <Text>
    <strong>{numOfSystems === 1 ? systemName : numOfSystems}</strong>{numOfSystems === 1 ? "" : " 
    systems"} will no longer be part of <strong>{groupName}</strong> and {numOfSystems === 1 ? "its" : 
    "their"} configuration will be impacted.
  </Text>
```

Reply:
Thank you. But still, I would like to stick to the current variant (see https://github.com/RedHatInsights/insights-inventory-frontend/pull/1974#discussion_r1294420391 comment)